### PR TITLE
Provides large page spec when loading state refs to make vault update.

### PR DESCRIPTION
**Pull request by joeldudleyr3**

### CODE REVIEW

File: node/src/main/kotlin/net/corda/node/services/vault/NodeVaultService.kt [hunk 1]
```diff
@@ -186,9 +186,11 @@ class NodeVaultService(
     }
 
     private fun loadStates(refs: Collection<StateRef>): Collection<StateAndRef<ContractState>> {
-        return if (refs.isNotEmpty())
-            queryBy<ContractState>(QueryCriteria.VaultQueryCriteria(stateRefs = refs.toList())).states
-        else emptySet()
+        return if (refs.isNotEmpty()) {
+            val stateRefQueryCriteria = QueryCriteria.VaultQueryCriteria(stateRefs = refs.toList())
+            val pageSpec = PageSpecification(MAX_PAGE_SIZE)
```
Comment (josecoll) [inline node/src/main/kotlin/net/corda/node/services/vault/NodeVaultService.kt]: This is not a good idea either as can potentially lead to JVM OOM exceptions.
We need to constrain the maximum number of allowable states in a transaction (should be tied into the Network Parameters maximum transaction size).

**Comment by josecoll** [comment]
Yes - and we should publish these upper bounds in our documentation.

**Comment by joeldudleyr3** [comment]
Cool. Are you capturing all this in the JIRA? Link to the JIRA in the issue once you're done and I'll close this.

**Comment by josecoll** [comment]
Please raise the JIRA as you have completed the PR (and initial investigation).

**Pull request closed without merging**